surfactant/relationships/dotnet_relationship.py

```diff
@@ -65,6 +66,58 @@ def establish_relationships(
+    # 2. Calling the import resolver set by the setDllImportResolver function
+    #    - a. Example using SetDllImportResolver: https://learn.microsoft.com/en-us/dotnet/standard/native-interop/native-library-loading
+    #    - b. Checks PInvoke's or Assembly's DefaultDllImportSearchPathsAttribute, then the assembly's directory, then LoadLibraryEx with LOAD_WITH_ALTERED_SEARCH_PATH flag
+    #       - This attribute has no effect on non-Windows platforms / Mono runtime
+    #       - https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.defaultdllimportsearchpathsattribute?view=net-7.0
+    #       - i. This has a "Paths" property which is a bitwise combination of paths specified in ii:
+    #       - ii. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportsearchpath?view=net-7.0
+    # 3. The active AssemblyLoadContext calls its LoadUnmanagedDll function (Default behavior is the same as AssemblyRef probing?)
+    #    - a. Can be overridden, but the default implementation returns IntPtr.Zero, which tells the runtime to load with its default policy.
+    #    - b. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext.loadunmanageddll?view=net-7.0
+    # 4. Run default unmanaged library probing logic by parsing *.deps.json probing properties
+    #    - a. If the json file isn't present, assume the calling assembly's directory contains the library
+    #    - b. https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/default-probing#unmanaged-native-library-probing
+    if "dotnetImplMap" in metadata:
+        for asmRef in metadata["dotnetImplMap"]:
+            if "Name" not in asmRef:
+                continue
+            refName = asmRef["Name"]
+            if is_absolute_path(refName):
+                relationships.extend(get_windows_pe_dependencies(sbom, software, [refName]))
+                continue
+            probedirs = []
+            if isinstance(software.installPath, Iterable):
+                for ipath in software.installPath:
+                    probedirs.append(pathlib.PureWindowsPath(ipath).parent.as_posix())
+            combinations = [refName]
+            if not (refName.endswith(".dll") or refName.endswith(".exe")):
+                combinations.append(f"{refName}.dll")
+            combinations.extend(
+                [
+                    f"{refName}.so",
+                    f"{refName}.dylib",
+                    f"lib{refName}.so",
+                    f"lib{refName}.dylib",
+                    f"lib{refName}",
+                ]
+            )
+            if ".so." in refName or refName.endswith(".so"):
+                combinations.extend(
+                    [
+                        f"{refName}.so.6",
+                        f"lib{refName}.so.6",
+                        f"{refName}.so.6.so",
+                        f"lib{refName}.so.6.so",
+                    ]
+                )
```

Comment:
Can you create a GitHub Issue about adding support for "fuzzy" dependency name matching using regular expressions? The `6` is actually a version related number for the shared library, so these should actually be matching against any number with a regex like `f"{refname}.so.[0-9]?"`. That's a fairly large overhaul of the functions we currently have for finding dependencies, so shouldn't be done right now.

These numbers in a shared library file name also tend to be associated with symlinks, so there is likely to be a plain (e.g.) `{refname}.so` file that will be matched because it symlinks to (e.g.) `{refname}.so.6`.

Reply:
#79 - backlog issue for adding fuzzy/regex dependency name matching support.